### PR TITLE
Remove deprecated spring properties

### PR DESCRIPTION
**Pull request by zeitlinger**
Based on 
~~- https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/10512~~
~~- https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/10408~~
~~- https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/10511~~
~~- https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/10530~~

Fixes #https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/10450

# Bug fix

`otel.propagators` were not read correctly: https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/10454/files#diff-70b5a0d550786f16703fb5d2338924d7d213793f610bc24a640bd5f8fa64def5R81 -> split off to https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/10559

# Breaking changes

All of these changes coverge towards the behavior of SDK auto config:

logging exporter
- `otel.exporter.logging.metrics.enabled` -> `otel.metrics.exporter=logging` (same for traces, metrics)
- `otel.exporter.logging.enabled=true` -> `otel.metrics.exporter=logging`, `otel.traces.exporter=logging`, `otel.logs.exporter=logging`

OTLP exporter
- `otel.exporter.otlp.metrics.enabled` -> `otel.metrics.exporter=otlp` (same for traces, metrics) (enabled by default)
- `otel.exporter.otlp.enabled` -> `otel.metrics.exporter=otlp`, `otel.traces.exporter=otlp`, `otel.logs.exporter=otlp`

Zipkin exporter
- `otel.exporter.zipkin.enabled` -> `otel.traces.exporter=zipkin`

Propagation
- `otel.propagation.type=tracecontext,baggage` -> `otel.propagators=tracecontext,baggage`

Resource
- `otel.springboot.resource.attributes` -> `otel.resource.attributes`

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -57,6 +57,7 @@ derby.log
 hs_err_pid*
 replay_pid*
 .attach_pid*
+AutoValue_*.java
```
Comment (trask) [inline .gitignore:60]: I think this is already handled by them being generated under `/build/`?

Comment (zeitlinger) [reply .gitignore:60]: no, I had it in the source dir - not sure why

File: instrumentation/spring/spring-boot-autoconfigure/build.gradle.kts [hunk 2]
```diff
@@ -37,7 +37,7 @@ dependencies {
   library("org.springframework.boot:spring-boot-starter-web:$springBootVersion")
   library("org.springframework.boot:spring-boot-starter-webflux:$springBootVersion")
 
-  compileOnly("io.opentelemetry:opentelemetry-sdk-extension-autoconfigure-spi")
+  implementation("io.opentelemetry:opentelemetry-sdk-extension-autoconfigure")
```
Comment (jeanbisutti) [inline instrumentation/spring/spring-boot-autoconfigure/build.gradle.kts:40]: It should not be `library("io.opentelemetry:opentelemetry-sdk-extension-autoconfigure")`, as the Spring dependencies? cc @laurit who should know

Comment (laurit) [reply instrumentation/spring/spring-boot-autoconfigure/build.gradle.kts:40]: `library` means `compileOnly` + `testImplementation`. Using `compileOnly` is important in `javaagent` modules, which this module is not. It is also used to support the latest dep test where we change the version for `library` dependencies to latest version. In my opinion here `implementation` is the correct choice as it is a required dependency for this module.

**Comment by laurit** [comment]
@zeitlinger usually before removing the properties we do a deprecation cycle, was there a decision to skip this for these? In the pr description could you provide the list of removed properties and their replacements. @trask usually list breaking changes in the changelog, having these available here would make it easier for him.

**Comment by zeitlinger** [comment]
> @zeitlinger usually before removing the properties we do a deprecation cycle, was there a decision to skip this for these? In the pr description could you provide the list of removed properties and their replacements. @trask usually list breaking changes in the changelog, having these available here would make it easier for him.

done :smile:

**Comment by zeitlinger** [comment]
> @zeitlinger usually before removing the properties we do a deprecation cycle, was there a decision to skip this for these? 

@trask and @jeanbisutti asked for the "fast track" because right now there are few users, but we want to make a push with the GA release

> In the pr description could you provide the list of removed properties and their replacements. @trask usually list breaking changes in the changelog, having these available here would make it easier for him.
 
 done 😄

**Review by jeanbisutti — APPROVED** [review]

**Comment by trask** [comment]
will merge after the release (and sorry @zeitlinger looks like lots of conflicts from splitting out the bug fix)

**Comment by zeitlinger** [comment]
> will merge after the release (and sorry @zeitlinger looks like lots of conflicts from splitting out the bug fix)

I just had to ignore my all changes from the other PR :smile: 

@trask ready to merge now

**Pull request merged**